### PR TITLE
Issue #864: Fixed excel sheet name length

### LINKED ISSUE #864: Transformer's "Write Data" fails on execute
It seems that we have a bug in the transformer feature that generates a data writer, at least in the case where that writer is an Excel writer. The generated _Create Excel spreadsheet_ analyzer will fail when executing the job.

### Steps to reproduce
1. Build job from scratch with i.e. _Customers_ datastore.
2. Link a _Country standardizer_ to the table.
3. Click _Write data_ and choose an Excel datastore.
4. Run job.

### Expected outcome
Job runs, data in new Excel spreadsheet.

### Actual outcome
Job fails, with varying Exceptions.

Sometimes
```
java.lang.NullPointerException
	at org.apache.metamodel.excel.ExcelUpdateCallback.getLastRowNum(ExcelUpdateCallback.java:131)
	at org.apache.metamodel.excel.ExcelUpdateCallback.createRow(ExcelUpdateCallback.java:125)
	at org.apache.metamodel.excel.ExcelInsertBuilder.execute(ExcelInsertBuilder.java:52)
	at org.datacleaner.output.AbstractMetaModelOutputWriter$1.run(AbstractMetaModelOutputWriter.java:94)
	at org.apache.metamodel.excel.ExcelDataContext.executeUpdate(ExcelDataContext.java:232)
	at org.datacleaner.output.AbstractMetaModelOutputWriter.flushBuffer(AbstractMetaModelOutputWriter.java:85)
	at org.datacleaner.output.AbstractMetaModelOutputWriter.addToBuffer(AbstractMetaModelOutputWriter.java:78)
	at org.datacleaner.output.MetaModelOutputRow.write(MetaModelOutputRow.java:62)
	at org.datacleaner.extension.output.AbstractOutputWriterAnalyzer.writeRow(AbstractOutputWriterAnalyzer.java:79)
	at org.datacleaner.extension.output.AbstractOutputWriterAnalyzer.run(AbstractOutputWriterAnalyzer.java:68)
	at org.datacleaner.job.runner.AnalyzerConsumer.consumeInternal(AnalyzerConsumer.java:71)
	at org.datacleaner.job.runner.AbstractRowProcessingConsumer.consume(AbstractRowProcessingConsumer.java:161)
	at org.datacleaner.job.runner.ConsumeRowHandlerDelegate.consume(ConsumeRowHandlerDelegate.java:67)
	at org.datacleaner.job.runner.ConsumeRowHandlerDelegate.processNext(ConsumeRowHandlerDelegate.java:90)
	at org.datacleaner.job.runner.TransformerConsumer.consumeInternal(TransformerConsumer.java:115)
	at org.datacleaner.job.runner.AbstractRowProcessingConsumer.consume(AbstractRowProcessingConsumer.java:161)
	at org.datacleaner.job.runner.ConsumeRowHandlerDelegate.consume(ConsumeRowHandlerDelegate.java:64)
	at org.datacleaner.job.runner.ConsumeRowHandlerDelegate.processNext(ConsumeRowHandlerDelegate.java:90)
	at org.datacleaner.job.runner.TransformerConsumer.consumeInternal(TransformerConsumer.java:115)
	at org.datacleaner.job.runner.AbstractRowProcessingConsumer.consume(AbstractRowProcessingConsumer.java:161)
	at org.datacleaner.job.runner.ConsumeRowHandlerDelegate.consume(ConsumeRowHandlerDelegate.java:64)
	at org.datacleaner.job.runner.ConsumeRowHandler.consumeRow(ConsumeRowHandler.java:146)
	at org.datacleaner.job.tasks.ConsumeRowTask.execute(ConsumeRowTask.java:51)
	at org.datacleaner.job.concurrent.TaskRunnable.run(TaskRunnable.java:61)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
```

And sometimes a NullPointerException with no stacktrace (Maybe available if JVM configured with ```-XX:-OmitStackTraceInFastThrow```.

If the file exists without the sheet, this exception will be thrown instead:
```
WARN  10:41:44.040 [analyzerbeans-pool1-thread-2] ErrorAwareAnalysisListener - Exception stack trace:
java.lang.ArrayIndexOutOfBoundsException: 11
	at org.apache.metamodel.data.AbstractRowBuilder.value(AbstractRowBuilder.java:108)
	at org.apache.metamodel.data.AbstractRowBuilder.value(AbstractRowBuilder.java:103)
	at org.datacleaner.output.AbstractMetaModelOutputWriter$1.run(AbstractMetaModelOutputWriter.java:92)
	at org.apache.metamodel.excel.ExcelDataContext.executeUpdate(ExcelDataContext.java:232)
	at org.datacleaner.output.AbstractMetaModelOutputWriter.flushBuffer(AbstractMetaModelOutputWriter.java:85)
	at org.datacleaner.output.AbstractMetaModelOutputWriter.addToBuffer(AbstractMetaModelOutputWriter.java:78)
	at org.datacleaner.output.MetaModelOutputRow.write(MetaModelOutputRow.java:62)
	at org.datacleaner.extension.output.AbstractOutputWriterAnalyzer.writeRow(AbstractOutputWriterAnalyzer.java:79)
	at org.datacleaner.extension.output.AbstractOutputWriterAnalyzer.run(AbstractOutputWriterAnalyzer.java:68)
	at org.datacleaner.job.runner.AnalyzerConsumer.consumeInternal(AnalyzerConsumer.java:71)
	at org.datacleaner.job.runner.AbstractRowProcessingConsumer.consume(AbstractRowProcessingConsumer.java:161)
	at org.datacleaner.job.runner.ConsumeRowHandlerDelegate.consume(ConsumeRowHandlerDelegate.java:67)
	at org.datacleaner.job.runner.ConsumeRowHandler.consumeRow(ConsumeRowHandler.java:146)
	at org.datacleaner.job.tasks.ConsumeRowTask.execute(ConsumeRowTask.java:51)
	at org.datacleaner.job.concurrent.TaskRunnable.run(TaskRunnable.java:61)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
```

Please note that none of this happens if the _Create Excel spreadsheet_ analyzer is created manually.

### PR DESCRIPTION
**Pull request by kaspersorensen**
Fixes #864 by enforcing max 31 chars [1]

[1] see http://stackoverflow.com/questions/3681868/is-there-a-limit-on-an-excel-worksheets-name-length for details on that restriction

### CODE REVIEW

**Comment by ClaudiaPHI** [comment]
LGTM!

File: components/writers/src/main/java/org/datacleaner/extension/output/CreateExcelSpreadsheetAnalyzer.java [hunk 1]
```diff
@@ -153,21 +161,24 @@ public void validate() {
...
     }
 
     private String fixSheetName(String sheet) {
-        for (char c : ILLEGAL_SHEET_CHARS) {
+        for (char c : SHEET_NAME_ILLEGAL_CHARS) {
             while (sheet.indexOf(c) != -1) {
                 sheet = sheet.replace(c, '-');
             }
         }
+        if (sheet.length() > SHEET_NAME_MAX_LENGTH) {
```
Comment (LosD) [inline components/writers/src/main/java/org/datacleaner/extension/output/CreateExcelSpreadsheetAnalyzer.java]: Can we, in addition to cutting it off when it is too long here, somehow make the generated sheet names shorter?

Comment (LosD) [inline components/writers/src/main/java/org/datacleaner/extension/output/CreateExcelSpreadsheetAnalyzer.java]: I realize that I should probably elaborate: Beyond removing "output-", which I guess is still a small part, I was thinking something along putting the datastore name into the file name (that would also improve the actual filename, right now it is the horribly generic "_DataCleaner-staging.xlsx_"), and only put the transformer name into the sheet name.

Comment (kaspersorensen) [inline components/writers/src/main/java/org/datacleaner/extension/output/CreateExcelSpreadsheetAnalyzer.java]: One word: No

Comment (kaspersorensen) [inline components/writers/src/main/java/org/datacleaner/extension/output/CreateExcelSpreadsheetAnalyzer.java]: Hehehehehe ;-)

Comment (kaspersorensen) [inline components/writers/src/main/java/org/datacleaner/extension/output/CreateExcelSpreadsheetAnalyzer.java]: Yes we could, but I don't care about it right now - this solves the issue ;-)

Comment (LosD) [inline components/writers/src/main/java/org/datacleaner/extension/output/CreateExcelSpreadsheetAnalyzer.java]: I'll create an issue, then we can decide later :)

**Pull request merged**